src/test/java/com/aws/greengrass/builtin/services/pubsub/PubSubIPCEventStreamAgentTest.java

```diff
@@ -421,23 +536,60 @@ void GIVEN_subscribed_consumer_WHEN_invalid_topic_THEN_throws() {
     }
 
     @Test
-    void GIVEN_subscribed_consumer_to_wildcard_WHEN_publish_to_subtopic_THEN_publishes_And_THEN_unsubscribes()
+    void GIVEN_subscribed_consumer_to_wildcard_and_no_receive_mode_WHEN_publish_to_subtopic_THEN_not_publishes_And_THEN_unsubscribes()
             throws InterruptedException {
         CountDownLatch countDownLatch = new CountDownLatch(1);
         Consumer<PublishEvent> consumer = getConsumer(countDownLatch);
+        // Default is RECEIVE_MESSAGES_FROM_OTHERS if mode is not set for wildcard topic
         pubSubIPCEventStreamAgent.subscribe(TEST_WILDCARD_TOPIC, consumer, TEST_SERVICE);
 
         assertEquals(1, pubSubIPCEventStreamAgent.getListeners().size());
         assertTrue(pubSubIPCEventStreamAgent.getListeners().containsKey(TEST_WILDCARD_TOPIC));
         assertEquals(1, pubSubIPCEventStreamAgent.getListeners().get(TEST_WILDCARD_TOPIC).size());
 
         pubSubIPCEventStreamAgent.publish("Test/A/Topic/B/C", "ABCDEF".getBytes(), TEST_SERVICE);
-        assertTrue(countDownLatch.await(10, TimeUnit.SECONDS));
+        assertFalse(countDownLatch.await(10, TimeUnit.SECONDS));
```

Comment:
you sure about this change?

Reply:
yeah it's the case where internal plugin subscribe() with a wild carded topic. For existing usage of subscribe(), there is no receiveMode set, so its receiveMode will be set to RECEIVE_MESSAGES_FROM_OTHERS for a wildcarded topic and the message will not be published

Reply:
looks fine - the next test case `GIVEN_subscribed_consumer_to_wildcard_and_receive_all_mode_WHEN_publish_to_subtopic_THEN_publishes_And_THEN_unsubscribes` is what this test was previously doing